Frontend/ui-customer/src/Screens/CustomerUtils/MyProfileScreen/Controller.purs

```diff
@@ -71,47 +75,83 @@ instance loggableAction :: Loggable Action where
       PopUpModal.ETextController act -> trackAppTextInput appId (getScreen MY_PROFILE_SCREEN) "popup_modal_action" "primary_edit_text"
       PopUpModal.CountDown arg1 arg2 arg3 arg4 -> trackAppScreenEvent appId (getScreen MY_PROFILE_SCREEN) "popup_modal_action" "countdown_updated"
     ReqDelAccount -> trackAppActionClick appId (getScreen MY_PROFILE_SCREEN) "in_screen" "delete_account_request"
-    
+    ShowOptions -> trackAppActionClick appId (getScreen MY_PROFILE_SCREEN) "in_screen" "open_gender_options_drop_down"
+    GenderSelected value -> trackAppActionClick appId (getScreen MY_PROFILE_SCREEN) "in_screen" "gender_selected"
+    EmailIDEditTextAction (PrimaryEditText.TextChanged id value) -> trackAppActionClick appId (getScreen MY_PROFILE_SCREEN) "edit_email_text_changed" "primary_edit_text"
+    NoAction -> trackAppActionClick appId (getScreen MY_PROFILE_SCREEN) "in_screen" "no_action"
+    AnimationEnd _ -> trackAppActionClick appId (getScreen MY_PROFILE_SCREEN) "show_options" "animation_end"
+
 data Action = GenericHeaderActionController GenericHeader.Action
             | BackPressed MyProfileScreenState
             | AfterRender
-            | EditProfile
+            | EditProfile (Maybe FieldType)
             | NameEditTextAction PrimaryEditText.Action
+            | EmailIDEditTextAction PrimaryEditText.Action
             | UpdateButtonAction PrimaryButton.Action
             | UserProfile GetProfileRes
             | PopUpModalAction PopUpModal.Action
             | AccountDeletedModalAction PopUpModal.Action
             | ReqDelAccount
+            | ShowOptions
+            | GenderSelected Gender
+            | NoAction
+            | AnimationEnd String
 
 data ScreenOutput = GoToHomeScreen | UpdateProfile MyProfileScreenState | DeleteAccount MyProfileScreenState | GoToHome
 eval :: Action -> MyProfileScreenState -> Eval Action ScreenOutput MyProfileScreenState
 
 eval (GenericHeaderActionController (GenericHeader.PrefixImgOnClick)) state = continueWithCmd state [do pure $ BackPressed state]
 
-eval (BackPressed backpressState) state = do 
-  if state.props.updateProfile then 
-    continue state { props { updateProfile = false } }
+eval (BackPressed backpressState) state = do
+  if state.props.updateProfile then do
+    _ <- pure $ hideKeyboardOnNavigation true
```

Comment:
BackPress should automatically hide the keyboard. It isnt the case right now?

Reply:
this doesn't work when the BackPressed action is triggered by the back_icon.